### ISSUE TITLE
Cannot start playing a new book after finishing a book.

### ISSUE THREAD

**Issue by FolkyHobbit** [PaulWoitaschek/Voice#708]
App-Version: 3.6.3
Phone Name: Moto E4 plus
Android Version: 7.1.1

After listening to a book to the end, the player is unable to start a new book. The play button changes into a pause button, in the home screen the icon appears at the new book, and in my notifications I get the new book (all as expected), but no audio starts playing and the time does not progress. Only workaround that I found is to kill the app from the settings. after this I am able to start the new book. 

Reproducing on my phone is easy and works every time: open a book, skip to the end and hit play. After this the player locks up. 

possibly related to #706 and #704?? They are also issues where the player locks up, but under different circumstances.

**Comment by holocronweaver** [comment]
Cannot reproduce using 3.6.3 on Android 8.1. 

If you have the Android SDK installed, can you run `adb logcat` while performing this operation and paste the resulting log lines here? If you phone is rooted you could alternatively use the CatLog app.

**Comment by FolkyHobbit** [comment]
I am not very familiar with the tools or android development, but I managed to get this, is this what you are looking for?
[logcat.txt](https://github.com/PaulWoitaschek/Voice/files/1863154/logcat.txt)

**Comment by holocronweaver** [comment]
Thanks! I see allot of errors related to the metadata extracting having trouble opening a MP3 audio file using DRM:

```
03-30 10:48:17.203   333   589 D StagefrightMetadataRetriever: fd=10,path=/storage/B888-1011/audiobook/Randall Munroe - What If Serious Scientific Answers to Absurd Hypothetical Questions (Unabridged)/What If Serious Scientific Answers to Absurd Hypothetical Questions (Unabridged) - 023.mp3
03-30 10:48:17.205   333   589 D DrmManagerClient(Native): DrmManagerClient(): mUniqueId=[4749]
03-30 10:48:17.205   328   328 D DrmManager(Native): pid = 11691 
03-30 10:48:17.205   328   328 D DrmManager(Native): pid_str = 11691
03-30 10:48:17.205   328   328 E DrmMtkUtil/DrmUtil: parseDcf: not dcf type, dcf version value [0]
03-30 10:48:17.205   328   328 E DrmMtkPlugIn: onOpenDecryptSession() : failed to parse dcf file.
03-30 10:48:17.207   333   589 D DrmManagerClient(Native): ~DrmManagerClient(): mUniqueId=[4749]
03-30 10:48:17.209   332 10693 D MediaExtractor: JB +MediaExtractor::Create
03-30 10:48:17.210   333   589 D DrmManagerClient(Native): DrmManagerClient(): mUniqueId=[7577]
03-30 10:48:17.211   328  3342 D DrmManager(Native): pid = 332 
03-30 10:48:17.211   328  3342 D DrmManager(Native): pid_str = 332
03-30 10:48:17.211   328  3342 E DrmMtkUtil/DrmUtil: parseDcf: not dcf type, dcf version value [0]
03-30 10:48:17.211   328  3342 E DrmMtkPlugIn: onOpenDecryptSession() : failed to parse dcf file.
03-30 10:48:17.213   333   589 D DrmManagerClient(Native): ~DrmManagerClient(): mUniqueId=[7577]
03-30 10:48:17.230   332 10693 D MPEG2PSExtractor: +SniffMPEGPS in
03-30 10:48:17.230   332 10693 D MPEG2PSExtractor: ********************* parsePackHeader can't find pack header start code *********************
03-30 10:48:17.241   333   589 E DrmMtkUtil/DrmUtil: checkDcf: not dcf type, dcf version value [0]
03-30 10:48:17.241   333   589 E Cta5File: Cta5File::isCta5NormalFile false, bad magic:
03-30 10:48:17.241   333   589 E Cta5File: Cta5File::isCta5MultimediaFile false, bad magic:
03-30 10:48:17.241   333   589 E StagefrightMetadataRetriever: Unable to instantiate an extractor for '10'.
03-30 10:48:17.241   333   589 D StagefrightMetadataRetriever: ~StagefrightMetadataRetriever()
03-30 10:48:17.405 11691 11721 E ExoPlayerImplInternal: Source error.
03-30 10:48:17.405 11691 11721 E ExoPlayerImplInternal: com.google.android.exoplayer2.source.UnrecognizedInputFormatException: None of the available extractors (MatroskaExtractor, FragmentedMp4Extractor, Mp4Extractor, Mp3Extractor, AdtsExtractor, Ac3Extractor, TsExtractor, FlvExtractor, OggExtractor, PsExtractor, WavExtractor, FlacExtractor) could read the stream.
03-30 10:48:17.405 11691 11721 E ExoPlayerImplInternal: 	at com.google.android.exoplayer2.source.ExtractorMediaPeriod$ExtractorHolder.selectExtractor(ExtractorMediaPeriod.java:784)
03-30 10:48:17.405 11691 11721 E ExoPlayerImplInternal: 	at com.google.android.exoplayer2.source.ExtractorMediaPeriod$ExtractingLoadable.load(ExtractorMediaPeriod.java:707)
03-30 10:48:17.405 11691 11721 E ExoPlayerImplInternal: 	at com.google.android.exoplayer2.upstream.Loader$LoadTask.run(Loader.java:315)
03-30 10:48:17.405 11691 11721 E ExoPlayerImplInternal: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
03-30 10:48:17.405 11691 11721 E ExoPlayerImplInternal: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
03-30 10:48:17.405 11691 11721 E ExoPlayerImplInternal: 	at java.lang.Thread.run(Thread.java:761)
...

```

Not sure if this is relevant to your issue though. Does this problem occur for any book, or does it only occur for certain ones? Maybe try this where the current and following book are both DRM free and see if the issue persists.

Did you reproduce the error while this logcat was being recorded? Do you know roughly at what time the error occurred so I can look at timestamps in that vicinity? If not, please repeat this logcat process, noting the time at which the error occurred.

Also you could provide me some files that cause this issue for you and I could try to reproduce locally.

**Comment by gwarser** [comment]
[logcat.zip](https://github.com/PaulWoitaschek/Voice/files/1865607/logcat.zip) three attempts.

I see nothing drm related in my logs. Metadata extracting errors happens on books listing - after first book ends, and I press back, before entering second book. Happens for all audiobooks (last 10?), but may be related to foobar2000 - files may be _preprocessed_ (tags/covers removed).

**Comment by holocronweaver** [comment]
Unfortunately those logs don't seem to indicate what is causing the issue.

I suspect PR #711 will fix your issue as it is related #706 where controls lock when editing a currently playing book, which uses the same code path a newly played book uses.

If you can provide me samples of the audiobooks that cause issues, I can test the fix for  you.

Alternatively I could provide you an APK to test with, but note that doing this will delete your empty your audiobook database and settings.

**Comment by rich123** [comment]
I am seeing the identical 'unable to play' on my phone, which is also an Moto E4 (just not the Plus model).

App-Version: 3.6.1 (from F-Droid)
Phone Name: Moto E4
Android Version: 7.1.1

I installed Voice to use to play podcasts, primarily due to the built in variable speed playback option.  If I let a podcast run to the end, then most often Voice will not start playing another track until after I've done a "force stop" in the Android "apps" page for Voice.

**Comment by gwarser** [comment]
Dupe https://github.com/PaulWoitaschek/Voice/issues/676

**Comment by PaulWoitaschek** [comment]
Should be fixed with the upcoming 4.0 release.

**Comment by MagTun** [comment]
While waiting for 4.0, here is the trick: let say you have just finish audiobook 1 and you want to listen to audiobook 2. Instead of going straight to audiobook 2, stay on audiobook 1, press play, wait one second, play pause, and then go to audiobook 2 and press play. By doing that, I no longer experienced the bug.
